polyply/tests/test_map_to_molecule.py

```diff
@@ -64,130 +64,130 @@ def test_add_blocks():
+        ; name nexcl.
+        PEO         1
+        ;
+        [ atoms ]
+        1  SN1a    1   PEO   CO1  1   0.000  45
+        [ moleculetype ]
+        ; name nexcl.
+        MIX         1
+        ;
+        [ atoms ]
+        1  SN1a    1   R1   C1  1   0.000  45
+        2  SC1     1   R1   C1  2   0.000  45
+        3  SN1a    2   R2   C1  1   0.000  45
+        4  SC1     2   R2   C1  2   0.000  45
+        5  SN1a    3   R3   C1  1   0.000  45
+        6  SC1     3   R3   C1  2   0.000  45
+        7  SN1a    4   R2   C1  1   0.000  45
+        8  SC1     4   R2   C1  2   0.000  45
+        [ bonds ]
+        ; back bone bonds
+        1  2   1   0.44 7000
+        1  3   1   0.37 7000
+        3  4   1   0.44 7000
+        3  5   1   0.37 7000
+        3  7   1   0.37 7000
+        5  6   1   0.44 7000
+        7  8   1   0.44 7000
+        """,
+        ["PEO", "MIX"],
+        [Interaction(atoms=(1, 2), parameters=['1', '0.44', '7000'], meta={}),
+         Interaction(atoms=(1, 3), parameters=['1', '0.37', '7000'], meta={}),
+         Interaction(atoms=(3, 4), parameters=['1', '0.44', '7000'], meta={}),
+         Interaction(atoms=(3, 5), parameters=['1', '0.37', '7000'], meta={}),
+         Interaction(atoms=(3, 7), parameters=['1', '0.37', '7000'], meta={}),
+         Interaction(atoms=(5, 6), parameters=['1', '0.44', '7000'], meta={}),
+         Interaction(atoms=(7, 8), parameters=['1', '0.44', '7000'], meta={})],
+        [(1, 2), (2, 3), (2, 4)],
+        5,
+        {0: "PEO", 1: "R1", 2: "R2", 3: "R3", 4: "R2"})
+        ))
+    def test_multiresidue_block(lines, monomers, bonds, edges, nnodes, resnames):
+        """
+        Test if a linear molecule of multiple residues is exanded,
+        correctly into a multi-block residue.
+        """
+        lines = textwrap.dedent(lines).splitlines()
+        ff = vermouth.forcefield.ForceField(name='test_ff')
+        polyply.src.polyply_parser.read_polyply(lines, ff)
+        meta_mol = MetaMolecule(name="test", force_field=ff)
+        meta_mol.add_monomer(0, monomers[0], [])
+        meta_mol.add_monomer(1, monomers[1], [(1,0)])
 
-#       new_meta_mol = polyply.src.map_to_molecule.MapToMolecule().run_molecule(meta_mol)
+        new_meta_mol = polyply.src.map_to_molecule.MapToMolecule(ff).run_molecule(meta_mol)
 
-#       assert new_meta_mol.molecule.interactions['bonds'] == bonds
-#       assert len(new_meta_mol.nodes) == nnodes
-#       assert list(new_meta_mol.edges) == edges
-#       assert nx.get_node_attributes(new_meta_mol, "resname") == resnames
-#       assert len(nx.get_node_attributes(new_meta_mol, "graph")) == nnodes
+        assert new_meta_mol.molecule.interactions['bonds'] == bonds
+        #assert len(new_meta_mol.nodes) == nnodes
+        #assert list(new_meta_mol.edges) == edges
+        #assert nx.get_node_attributes(new_meta_mol, "resname") == resnames
+        #assert len(nx.get_node_attributes(new_meta_mol, "graph")) == nnodes
```

Comment:
?

Reply:
this is a leftover; within the new concept the meta-molecule is not expanded if the a multi-residue block is referred to by block-name as single residue. Previously this was required because the nodes were coupled to the resids. But as that has been resolved there is no point, as we also don't make default edges anymore.

Reply:
In fact I rewrote the tests and extended them; found a bug and now it should be good. But can you have a look at the test cases and see if you spot some missing combination or so